zipkin-autoconfigure/metrics-prometheus/src/main/java/zipkin/autoconfigure/prometheus/ZipkinPrometheusMetricsAutoConfiguration.java

```diff
@@ -14,78 +14,81 @@
 
 package zipkin.autoconfigure.prometheus;
 
-import io.prometheus.client.Histogram;
-import io.prometheus.client.hotspot.DefaultExports;
-import io.prometheus.client.spring.boot.EnablePrometheusEndpoint;
-import io.prometheus.client.spring.boot.EnableSpringBootMetricsCollector;
-import io.prometheus.client.spring.web.EnablePrometheusTiming;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Timer;
 import io.undertow.server.HandlerWrapper;
 import io.undertow.server.HttpHandler;
 import io.undertow.server.HttpServerExchange;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.boot.context.embedded.undertow.UndertowDeploymentInfoCustomizer;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
-@EnablePrometheusEndpoint
-@EnableSpringBootMetricsCollector
-@EnablePrometheusTiming
 @Configuration
-class ZipkinPrometheusMetricsAutoConfiguration {
-  // needs to be a JVM singleton
-  static final Histogram http_request_duration_seconds = Histogram.build()
-    .labelNames("path", "method")
-    .help("Response time histogram")
-    .name("http_request_duration_seconds")
-    .register();
+public class ZipkinPrometheusMetricsAutoConfiguration {
+  private MeterRegistry registry;
 
-  ZipkinPrometheusMetricsAutoConfiguration() {
-    DefaultExports.initialize();
-  }
-
-  @Bean Histogram httpRequestDuration() {
-    return http_request_duration_seconds;
+  ZipkinPrometheusMetricsAutoConfiguration(MeterRegistry registry) {
```

Comment:
is this the same as `@Autowired` on a field? (I mean does spring happily inject this?)

Reply:
Yeah spring inject this bean.